client/src/api/v3/boost.ts

```diff
@@ -1,28 +1,109 @@
 import { emit } from 'api/common/socket';
 import { BoostConfigType } from 'types/boost';
+import toast from 'react-hot-toast';
+
+type payloadAction = 'upload' | 'delete';
 
 /**
- * Send the content of the given configuration file on `boost/configs/action` over MQTT
- *
+ * Send configuration status over MQTT on topic 'boost/configs/action'
+ * 
+ * @param actionType represents whether the config is being uploaded or deleted
  * @param type the type of the configuration being sent
- * @param configFile file containing the configuration content
+ * @param configContent configuration content
+ */
+function sendConfig(actionType: payloadAction, type: BoostConfigType, configContent: string | null) {
+  const topic = 'send-config';
+  const payload = {
+    action: actionType,
+    configType: type,
+    content: configContent,
+  };
+  // To check things work without needing the server started
+  console.log(payload);
+  emit(topic, JSON.stringify(payload));
+}
+
+/**
+ * Read content from the given file and send it on `boost/configs/action` over MQTT. 
+ * If the content contains more than one config (i.e. `type` is 'all'), the content is
+ * split into the different configurations before sending.
+ *
+ * @param type the type of the configuration
+ * @param configFile file containing content of the configuration
+ * @param displayErr function to display error if uploaded config is not correct
+ * @param configExist function to check if a given config name has already been used
  */
 export default function uploadConfig(
   type: BoostConfigType,
   configFile: File,
+  displayErr: (message: string) => void,
+  configExist: (type: BoostConfigType, name: string) => boolean,
 ) {
-  const topic = 'send-config';
   const reader = new FileReader();
+  const possibleConfig = ['powerPlan', 'rider', 'track', 'bike'];
 
   // Called when FileReader has completed reading a file
   reader.onload = () => {
-    console.log(reader.result);
-    const payload = {
-      action: 'upload',
-      configType: type,
-      content: reader.result,
+    const fileContent = reader.result as string;
+    if (type === 'all') {
```

Comment:
This function is quite large. To make it more manageable, you could make two new functions, one to handle `type === 'all'`, and the other to handle a single config. We'll also have a few less levels of indentation, making all the if statements easier to follow.

Reply:
Just wanted to add that in situations like this, I think it would be more understandable to keep the code as is (instead of making new single use functions), and instead maybe use `return` to reduce the indentation level. 

I think the overall amount of code can also be reduced by some of the suggestions given below.